client/src/components/Profile/EmploymentCard.tsx

```diff
@@ -0,0 +1,159 @@
+import isEqual from 'lodash/isEqual';
+import CommonCardWithSettingsModal from './CommonCardWithSettingsModal';
+import { EmploymentRecordDto, UpdateProfileInfoDto } from 'api';
+import EmploymentHistoryFormItem from './EmploymentHistory/EmploymentHistoryFormItem';
+import EmploymentHistoryDisplayItem from './EmploymentHistory/EmploymentHistoryDisplayItem';
+import dayjs from 'dayjs';
+
+export type EmploymentRecordFormItem = Omit<EmploymentRecordDto, 'dateFrom' | 'dateTo'> & {
+  dateFrom: dayjs.Dayjs;
+  dateTo: dayjs.Dayjs;
+};
+
+type Props = {
+  data: EmploymentRecordDto[];
+  isEditingModeEnabled: boolean;
+  updateProfile: (data: UpdateProfileInfoDto) => Promise<boolean>;
+};
+
+const dateFormat = 'YYYY-MM-DD';
+
+const defaultEmploymentRecord: EmploymentRecordDto = {
+  title: '',
+  dateFrom: '',
+  dateTo: '',
+  toPresent: false,
+  companyName: '',
+  officeLocation: undefined,
+};
+
+const EmploymentCard = ({ isEditingModeEnabled, data, updateProfile }: Props) => {
+  const [form] = Form.useForm<{ employmentHistory: EmploymentRecordFormItem[] }>();
+  const values = Form.useWatch([], form);
+
+  const employmentRecordFormItemToDto = (employment: EmploymentRecordFormItem): EmploymentRecordDto => ({
+    ...employment,
+    dateFrom: employment.dateFrom.format(dateFormat),
+    dateTo: employment.toPresent ? '' : employment.dateTo.format(dateFormat),
+  });
+
+  const employmentRecordDtoToFormItem = (employment: EmploymentRecordDto): EmploymentRecordFormItem => ({
+    ...employment,
+    dateFrom: employment.dateFrom ? dayjs(employment.dateFrom) : dayjs(),
+    dateTo: employment.dateTo ? dayjs(employment.dateTo) : dayjs(),
+  });
+
+  const [displayEmployments, setDisplayEmployments] = useState<EmploymentRecordFormItem[]>(
+    data.map(employmentRecordDtoToFormItem),
+  );
+  const [employments, setEmployments] = useState<EmploymentRecordFormItem[]>(displayEmployments);
+  const [isSaveDisabled, setIsSaveDisabled] = useState(true);
+  const [isFormValid, setIsFormValid] = useState(false);
+  const isAddDisabled = useMemo(() => !!employments.length && !isFormValid, [employments, isFormValid]);
+
+  useEffect(() => {
+    if (!values || !Object.values(values).every(Boolean)) {
+      return;
+    }
+
+    form.validateFields({ validateOnly: true }).then(
+      () => {
+        setIsFormValid(true);
+      },
+      () => {
+        setIsFormValid(false);
+      },
+    );
+  }, [values]);
+
+  useEffect(() => {
+    const readyToUpdate = !isEqual(displayEmployments, employments) && isFormValid;
+    setIsSaveDisabled(!readyToUpdate);
+  }, [displayEmployments, employments, isFormValid]);
+
+  const handleSave = async () => {
+    const employmentHistory = employments;
+    const isUpdated = await updateProfile({
+      employmentHistory: employmentHistory.map(employmentRecordFormItemToDto),
+    });
+
+    if (!isUpdated) {
+      return;
+    }
+
+    setDisplayEmployments(employments);
```

Comment:
related to `EmploymentHistoryDisplayItem` comment
So after save, if I don't reload the page I won't see employment from-present-duration, because it relies on `dateTo` to be truthy, 

this could solve it:
```js
setDisplayEmployments(employments.map(( item ) => ({...item, dateTo: item.dateTo ? dayjs(item.dateTo) : dayjs() })));
```

but IMHO the way you set `dateTo` on `employmentRecordDtoToFormItem` seams not right, if it's not set, it should stay not set
![image](https://github.com/rolling-scopes/rsschool-app/assets/67139199/5d033be6-6faa-4fc8-ba68-5e1799725b45) 
![image](https://github.com/rolling-scopes/rsschool-app/assets/67139199/e00f7b22-3309-4ea0-9e86-fa9c18275911)

better to handle it on `EmploymentHistoryDisplayItem`

Reply:
> this could solve it:
> 
> ```js
> setDisplayEmployments(employments.map(( item ) => ({...item, dateTo: item.dateTo ? dayjs(item.dateTo) : dayjs() })));
> ```

Yeah, I had such solution before somewhere, but for some reason I get rid of it.
Thanks I will double check this issue